hadoop-ozone/recon/src/test/java/org/apache/hadoop/ozone/recon/api/TestEndpoints.java

```diff
@@ -305,4 +419,39 @@ public void testGetPipelines() throws Exception {
       return (pipelineMetadata1.getContainers() == 1);
     });
   }
+
+  @Test
+  public void testGetClusterState() throws Exception {
+    Response response = clusterStateEndpoint.getClusterState();
+    ClusterStateResponse clusterStateResponse =
+        (ClusterStateResponse) response.getEntity();
+
+    Assert.assertEquals(1, clusterStateResponse.getPipelines());
+    Assert.assertEquals(2, clusterStateResponse.getVolumes());
+    Assert.assertEquals(2, clusterStateResponse.getBuckets());
+    Assert.assertEquals(3, clusterStateResponse.getKeys());
+    Assert.assertEquals(2, clusterStateResponse.getTotalDatanodes());
+    Assert.assertEquals(2, clusterStateResponse.getHealthyDatanodes());
+
+    DatanodeStorageReport storageReport =
+        clusterStateResponse.getStorageReport();
+
+
+    // if container report is processed first, and pipeline does not exist
+    // then container is not added until the next container report is processed
+    SCMHeartbeatRequestProto heartbeatRequestProto =
+        SCMHeartbeatRequestProto.newBuilder()
+            .setContainerReport(containerReportsProto)
+            .setDatanodeDetails(datanodeDetailsProto)
+            .build();
+    reconScm.getDatanodeProtocolServer()
+        .sendHeartbeat(heartbeatRequestProto);
+
+    LambdaTestUtils.await(30000, 2000, () -> {
```

Comment:
Line 440 - 454 has some some repetitive elements in all 3 tests. Only the condition is different. Maybe we can use a helper function to do this and remove duplicate code.

`  private void waitAndCheckConditionAfterHeartbeat(Callable<Boolean> check) throws Exception {
    // if container report is processed first, and pipeline does not exist
    // then container is not added until the next container report is processed
    SCMHeartbeatRequestProto heartbeatRequestProto =
        SCMHeartbeatRequestProto.newBuilder()
            .setContainerReport(containerReportsProto)
            .setDatanodeDetails(datanodeDetailsProto)
            .build();
    reconScm.getDatanodeProtocolServer()
        .sendHeartbeat(heartbeatRequestProto);
    LambdaTestUtils.await(30000, 2000, check);
  } `

It can be invoked by
`    waitAndCheckConditionAfterHeartbeat(() -> {
      Response response1 = clusterStateEndpoint.getClusterState();
      ClusterStateResponse clusterStateResponse1 =
          (ClusterStateResponse) response1.getEntity();
      return (clusterStateResponse1.getContainers() == 1);
    });`

I am OK to do this in the next JIRA if needed.

Reply:
Agreed. I can take care of this refactor in another JIRA.

Reply:
Thanks!

Reply:
Refactored as per the suggestion in the latest commit.